FirebaseRemoteConfig/Sources/RCNConfigRealtime.m

```diff
@@ -552,18 +562,30 @@ - (void)URLSession:(NSURLSession *)session
   }
 }
 
+/// Check if response code is retryable
+- (bool)isStatusCodeRetryable:(NSInteger)statusCode {
+  return statusCode == kRCNFetchResponseHTTPStatusClientTimeout ||
+         statusCode == kRCNFetchResponseHTTPStatusTooManyRequests ||
+         statusCode == kRCNFetchResponseHTTPStatusCodeServiceUnavailable ||
+         statusCode == kRCNFetchResponseHTTPStatusCodeBadGateway ||
+         statusCode == kRCNFetchResponseHTTPStatusCodeGatewayTimeout;
+}
+
 /// Delegate to handle initial reply from the server
 - (void)URLSession:(NSURLSession *)session
               dataTask:(NSURLSessionDataTask *)dataTask
     didReceiveResponse:(NSURLResponse *)response
      completionHandler:(void (^)(NSURLSessionResponseDisposition disposition))completionHandler {
   _isRequestInProgress = false;
   NSHTTPURLResponse *_httpURLResponse = (NSHTTPURLResponse *)response;
-  if ([_httpURLResponse statusCode] != 200) {
+  NSInteger statusCode = [_httpURLResponse statusCode];
+  if (statusCode != kRCNFetchResponseHTTPStatusOk) {
     [self pauseRealtimeStream];
-    [self retryHTTPConnection];
+    if ([self isStatusCodeRetryable:statusCode]) {
```

Comment:
for cases where we don't retry, should we return an error to the client?

Reply:
Yeah it'd make sense to return a StreamError if we can't retry, updated.